src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1969,8 +1972,21 @@
                        value op)))))
 
             [["null?" column-name value]]
-            (let [cinfo (get-in query-rec [:projections column-name])]
-              (map->NullExpression {:column cinfo :null? value}))
+            (let [maybe-dotted-path (str/split column-name #"\.")
+                  cinfo (get-in query-rec [:projections (first maybe-dotted-path)])]
+
+              ;; only matches inventory queries with a '(fs.stable||fs.volatile)' :field value
+              ;; that have a dotted path query like: facts.foo.bar
+              (if (and (= :queryable-json (:type cinfo))
+                       (h/raw? (:field cinfo))
```

Comment:
Does this introduce the requirement that queryable json fields must be raw or was that already in place?

I worry that the raw check will lead to us re-introducing this bug if we try to make a non-raw field queryable json.